tests/test_operation.py

```diff
@@ -794,3 +794,19 @@ def test_basis_state_decomposition(self):
 
         assert rec.queue[2].name == "PauliX"
         assert rec.queue[2].wires == [3]
+
+    def test_qubit_state_vector_decomposition(self):
+        """Test the decomposition of QubitStateVector is equivalent
+        to the MottonenStatePreparation template"""
+        state = np.array([1/2, 1j/np.sqrt(2), 0, -1/2])
+
+        with qml.utils.OperationRecorder() as rec1:
+            qml.QubitStateVector.decomposition(state, wires=[0, 1])
+
+        with qml.utils.OperationRecorder() as rec2:
+            qml.templates.state_preparations.MottonenStatePreparation(state, wires=[0, 1])
+
+        for op1, op2 in zip(rec1.queue, rec2.queue):
+            assert op1.name == op2.name
+            assert op1.wires == op2.wires
+            assert op1.params == op2.params
```

Comment:
```suggestion
            assert op1.params == op2.params

```

Reply:
Is it worth also adding an integration test here? For example, for a mocked default.qubit with no `QubitStateVector`, and ensure that the prepared state is correct?

Reply:
Actually I think now that just testing that `BasisStateDecomposition` and `MottonenStatePreparation` are called is wholly sufficient. The state preparation templates and the logic of decomposing when the operation is not supported are properly tested elsewhere.